module/VuFind/src/VuFind/Recommend/ConsortialVuFind.php

```diff
@@ -0,0 +1,242 @@
+
+    /**
+     * Number of results to show
+     *
+     * @var int
+     */
+    protected $limit = 5;
+
+    /**
+     * Connection to consortial VuFind API
+     *
+     * @var Connection
+     */
+    protected $connection;
+
+    /**
+     * ConsortialVuFind.ini configuration
+     *
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * Base URL of a search results page
+     *
+     * @var string
+     */
+    protected $resultsBaseUrl = null;
+
+    /**
+     * Base URL of a record page
+     *
+     * @var string
+     */
+    protected $recordBaseUrl = null;
+
+    /**
+     * Any filters used in the search
+     *
+     * @var array
+     */
+    protected $searchFilters = [];
+
+    /**
+     * Boolean indicating if at least the minimal required configuration is present
+     *
+     * @var bool
+     */
+    protected $hasMinimumConfig = false;
+
+    /**
+     * Query string from the original search results
+     *
+     * @var string
+     */
+    protected $queryString;
+
+    /**
+     * Constructor
+     *
+     * @param Config     $config     ConsortialVuFind.ini configuration
+     * @param Connection $connection Connection to consortial VuFind API
+     */
+    public function __construct(
+        Config $config,
+        Connection $connection
+    ) {
+        $this->config = $config;
+        $this->connection = $connection;
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        $settings = explode(':', $settings);
+        $this->requestParam = !empty($settings[0]) ? $settings[0] : $this->requestParam;
+        $this->limit
+            = (isset($settings[1]) && is_numeric($settings[1]) && $settings[1] > 0)
+            ? intval($settings[1]) : $this->limit;
+        $configSectionName = $settings[2] ?? 'ReShare';
+
+        // Read config file
+        $configSection = $this->config->get($configSectionName);
+        if ($configSection) {
+            $this->resultsBaseUrl = $configSection->results_base_url;
+            $this->recordBaseUrl = $configSection->record_base_url;
+            $this->searchFilters = $configSection->filters?->toArray() ?? [];
+
+            // Configure connection
+            $this->connection->setBaseUrl($configSection->api_base_url);
+
+            // Confirm that required configuration is present
+            $this->hasMinimumConfig = $this->resultsBaseUrl
+                && $this->recordBaseUrl
+                && $configSection->api_base_url;
+            if (!$this->hasMinimumConfig) {
+                $this->logError("Required configuration missing in '$configSectionName'
+                    section of ConsortialVuFind.ini.");
+            }
+        } else {
+            $this->logError("No section found '$configSectionName' in ConsortialVuFind.ini.");
```

Comment:
Nitpick, but I'd either add a colon after "found" or rephrase to `"'$configSectionName' section not found in ConsortialVuFind.ini."`.

Reply:
I agree, better to have the variable at the beginning or end.  Rephrased via the second option.